src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1363,7 +1364,8 @@ def _set_up_defaults(
 
     _infer_existing_custom_location_or_extension(cmd, name, location, resource_group, env, custom_location, extension)
 
-    _get_acr_from_image(cmd, app)
+    if not is_registry_server_params_set:
```

Comment:
I notice another issue about ACR image, if the ACR is anonymousPullEnabled, refer: https://learn.microsoft.com/en-us/azure/container-registry/anonymous-pull-access, no need to input `registry-server` and `registry-password`.

You can have a try with command `az containerapp create`:
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/dc7d2486-84d0-42e0-be26-5ea9dff50a6a)

Reply:
Currently with `az containerapp up` it will failed with error:
Failed to retrieve credentials for container registry. Please provide the registry username and password